packages/modules/src/lib/replaceArticleCount.tsx

```diff
@@ -9,10 +10,13 @@ export const replaceArticleCountWithLink = (
     tracking?: OphanTracking,
 ): Array<JSX.Element> => {
     const nextWords: Array<string | null> = [];
-    const subbedText = text.replace(/%%ARTICLE_COUNT%%( \w+)?/g, (_, nextWord) => {
-        nextWords.push(nextWord);
-        return '%%ARTICLE_COUNT_AND_NEXT_WORD%%';
-    });
+    const subbedText = text.replace(
+        RegExp(`${ARTICLE_COUNT_TEMPLATE}( \w+)?`, 'g'),
+        (_, nextWord) => {
+            nextWords.push(nextWord);
+            return '%%ARTICLE_COUNT_AND_NEXT_WORD%%';
+        },
+    );
```

Comment:
Could we write a test for this `subbedText` logic?  I feel like something with Regex such as this is ripe for testing, particularly after a refactor.

Reply:
Good shout @GHaberis, I'll add some unit tests this afternoon.

Reply:
I believe there's a difference between the result of the old "regular expression literal" implementation and the new `RegExp` constructor function implementation:


```
const ARTICLE_COUNT_TEMPLATE = '%%ARTICLE_COUNT%%';
const text = "Congratulations on being one of our top readers globally - you've read %%ARTICLE_COUNT%% articles in the last year."

console.log('old match', text.match(/%%ARTICLE_COUNT%%( \w+)?/g)) 
// OLD RESULT: ["%%ARTICLE_COUNT%% articles"]

console.log('new match', text.match(RegExp(`${ARTICLE_COUNT_TEMPLATE}( \w+)?`, 'g'))) 
// NEW RESULT: ["%%ARTICLE_COUNT%%"]
```

The `RegExp` constructor function implementation is not returning the match and the following word, in this case "articles"

When using the `RegExp` constructor function, the normal string escape rules (preceding special characters with \ when included in a string) are necessary. so we need an additional escape rule in `( \w+)`, so it becomes: `( \\w+)`. Doing this will mean the new `RegExp` returns the the match and the following word as before.

I've tested this at the attached JS sandox: https://jsfiddle.net/Lt45moub/2/